### ISSUE TITLE
Sysmon export logs to CSV | JSON | XML

### ISSUE THREAD

**Issue by kont45** [SwiftOnSecurity/sysmon-config#70]
Hello folks, 
Someone explain me how to export sysmon logs evtx to another format like xml, json or csv? I would like to learn more about malicious activities and harmful files, but reading original logs in Windows Event is a nightmare. 

I would be nice to export logs to friendly logs and parse logs, create a tree image path, PID or others relationships. It would be easier to read and analyse logs. 

I do not want to use Splunk or another remote logs server for sysmon. I would like to keep all data on same localhost.

**Comment by davebremer** [comment]
ok ... keep in mind that I'm a beginner - but I've had a couple of attempts which might be useful to get you started and kick you off using powershell. I started with a set of scripts that would export to csv  [Export-SysmonLogs](https://github.com/davebremer/Export-SysmonLogs). It was hard coded for each event type

Recently I had a crack of exporting them to a SQLIte file - easy to transport and I think might be useful to analyse [export-SysmonDB](https://github.com/davebremer/Export-SysmonDB). Sloowwwww - I really should use the bulkupload.

Basically (in the SysmonDB one)  - I'm using `(Get-WinEvent -ListProvider "Microsoft-Windows-Sysmon" ).Events` to get the schema of the various events and create a table for each event type. Then I go through and put each log into the appropriate table.

At least - that's the idea. Have a crack - there's got to be a better way :-)

**Comment by vavarachen** [comment]
I too dislike the XML based evtx format, which is why I developed evtx2json
<https://github.com/vavarachen/evtx2json>.  Keep in mind, that is not an
easy conversion, so the find JSON format is a bit odd.

V

**Comment by SecurityYam-DZ** [comment]
You can create a PowerShell script to do it and run it as a scheduled task. Something like the below should work.

```
$start = (Get-Date).addMinutes(-60)
$logs = Get-WinEvent -filterhashtable @{logname="Microsoft-Windows-Sysmon/Operational"; starttime = $start}
$logs | Export-Csv -Path [INSERT PATH]
```

What you can then do with that is use the Group-Object cmdlet to see some interesting things:

```
#Groups events by ID and sort them by occurence
$logs | Group-Object -property id | Sort-Object -property count
```

You'll notice that one of the most frequently occurring event is event ID 1, which usually contains an entry with the source image of the process that was called. If you want to see the groupings for images that were called, you can do this:

```
#Assign all events that match event ID 1 to variable $1
$1 = $logs | Where-Object {$_.id -eq 1}

#Find the strings that start with "Image", group them, and then sort them by occurrence
#1 | Select-String "^Image" | Group-Object | Sort-Object -property count
```

**Comment by kont45** [comment]
Thank you @SecurityYam-DZ. Your solution is mostly easy. Mostly... :) @davebremer. 
Guys, you must know, I'm completely beginner with powershell and sysmon. To me it is not easy create a complicated task and extract needed data.

@SecurityYam-DZ Can you lead me by hand and provide ready-made scripts to sort CSV into appropriate columns that are useful for analysing malware? For example, commands run in powershell by malware, loaded libraries, created new files, deleted, modified and read. It would be ideal to display it in the form of a tree of processes. Maybe other users will benefit from this and you can attach it to the official beginner's repository.

**Comment by SecurityYam-DZ** [comment]
Hi, @kont45 . 

I can look into putting this code into functions, but that would just involve copying and pasting the code that I sent you already into functions. 

There are also TechNet pages published by Microsoft that explain the usage of all their cmdlets if you Google them. Below are some cmdlets that I've found most relevant when working with Sysmon

- Get-WinEvent
- Group-Object
- Sort-Object
- Where-Object
- Export-Csv
- Select-String

In response to your second paragraph: Unfortunately, I don't have any ready-made scripts for analyzing Sysmon events. We've mostly been focusing on preventative controls and haven't really had a chance to refine our processes involving Sysmon yet. Creating a library of tools to do this is no easy undertaking that I may look at in the future. I'll possibly respond to this thread later when I've had some free time to work on this.

**Comment by kont45** [comment]
@SecurityYam-DZ I'll be grateful if you can do something soon.

**Comment by MikeLim7** [comment]
@kont45 check out Sysmon View at https://github.com/nshalabi/SysmonTools.

**Comment by cyberkryption** [comment]
Keeping the logs on the localhost is considered a bad security oractice
what if an attacker eeletes them? How would alert on them in a non reactive
way?

I suggest you check out graylog or elk  plus there is an amazon  version
called open distro, you could run a small vm or hardware such odroid h2

Cyberkryption

**Comment by MikeLim7** [comment]
Agree with @cyberkryption if this is for production. @kont45 , you can try Logging Make Easy, where they use Sysmon and push logs to ELK. Install scripts and docker images are provided.
https://github.com/ukncsc/lme

**Comment by kont45** [comment]
@cyberkryption @MikeLim7 The problem is focused on the analysis of malicious files for one PC. Keeping external server with logs does not make sense for me.

**Comment by MikeLim7** [comment]
Sysmon View should serve your needs.  We can close this issue.
